src/core/featuremodel.cpp

```diff
@@ -503,7 +503,7 @@ void FeatureModel::applyGeometry()
   // Extra steps to insure a valid geometry
   geometry = geometry.makeValid();
   QgsGeometry deduplicatedGeometry = geometry;
-  deduplicatedGeometry.removeDuplicateNodes( 7 );
+  deduplicatedGeometry.removeDuplicateNodes( 0.00000001 );
```

Comment:
If the layer has configured stng smaller in the geometry options that should be used

Reply:
@m-kuhn , good point.

Reply:
Actually this will be done automatically. So if it's configured let's skip it here altogether.

Reply:
@m-kuhn , done.